### DISCUSSION TITLE
Working with mobile React-Native application

### DISCUSSION THREAD

**Discussion by sarrahadjayache** [keycloakify/keycloakify#198] [Q&A]
Hello! I am a beginner to keycloak and currently i am working on integrating it in our professional project. 
for react js I found keyclokify documentation which looks very awesome and very helpful. But i'd like to know if it is possible to work with it on a react-native project for a mobile app.
I can't find for now any clear way  to do it in react-native.

Thanks in advance 😁

**Comment by garronej** [comment]
Hi Sarra,  

Login and register pages are generated on the server by Keycloak.  (Like if it where PHP pages, it's really FreeMarker)
You can't embed the login and register pages within you React Native app.  
It's not a limitation specific to Keycloakify, it's just not how openidconnect works.  

So you'll have to build your Keycloak theme using React for the web but after that you can integrate 
Keycloak to your React Native app.  
Webview components are going to be used when user login or register, that's all.  

My advice to you is to try and integrate Keycloak with your React Native application.  
You can have a look at this project, for example https://github.com/react-keycloak/react-native-keycloak (I don’t know what it's worth)  

Once you got this working, you'll be able to customize the look and feel of the login and register pages using Keycloakify but you will 
code the Login and Register page using React for the web, the code for your Keycloak theme will be a separate project from your main app.  

If you do it well it should be transparent for the user.  

Good luck

**Reply by sarrahadjayache** [reply]
Thank you 🙏

**Reply by abhi-ky-work** [reply]
Hi @sarrahadjayache , were you able to integrate it in React-Native App ?

**Comment by Devanshuji** [comment]
Hey @sarrahadjayache did you found any solution to integrating it in React-Native app
